### PR TITLE
magic-button can generate slugs now

### PR DESCRIPTION
**Pull request by yoshokatana**
magic-button should have a general slugification transform

### CODE REVIEW

File: behaviors/magic-button.js [hunk 1]
```diff
@@ -30,10 +43,46 @@ const _ = require('lodash'),
...
+      });
     }
   };
 
 /**
+ * Removes all unicode from string
+ * @param {string} str
+ * @returns {string}
+ */
+function stripUnicode(str) {
```
Comment (davidandrewfrey) [inline behaviors/magic-button.js]: Do we have a more Util-y place for this guy to live? Could see this being handy elsewhere.

File: behaviors/magic-button.js [hunk 2]
```diff
@@ -30,10 +43,46 @@ const _ = require('lodash'),
...
+  return str.replace(/[^A-Za-z 0-9\.,\?!@#\$%\^&\*\(\)-_=\+;:<>\/\\\|\}\{\[\]~]*/g, '');
+}
+
+/**
+ * remove all html stuff from a string
+ * @param {string} str
+ * @returns {string}
+ */
+function toPlainText(str) {
```
Comment (davidandrewfrey) [inline behaviors/magic-button.js]: Same Util comment for this function too.

File: behaviors/magic-button.js (hunk 1 above)
Comment (yoshokatana) [inline behaviors/magic-button.js]: not really. if we end up using it anywhere else, I'll think about moving it

File: behaviors/magic-button.js (hunk 2 above)
Comment (yoshokatana) [inline behaviors/magic-button.js]: same answer :p

File: behaviors/magic-button.js [hunk 3]
```diff
@@ -30,10 +43,46 @@ const _ = require('lodash'),
...
+  // coerce all text into a string. Undefined stuff is just an empty string
+  if (!_.isString(str)) {
```
Comment (davidandrewfrey) [inline behaviors/magic-button.js]: Our preference is to skip ternary operators for things like this?

Comment (yoshokatana) [inline behaviors/magic-button.js]: yeah, this is clearer

File: behaviors/magic-button.js [hunk 4]
```diff
@@ -122,10 +172,12 @@ function getProperty(property) {
  * do magic on click ☆.。.:*・°☆.。.:*・°☆.。.:*・°☆.。.:*・°☆
  * @param {MouseEvent} e
  * @param {object} bindings
+ * @param {Element} [testEl] for testing, we pass a stubbed element in rather than the event
+ * note: when this function is called from the binding, testEl is undefined
  * @returns {Promise|undefined}
  */
-function doMagic(e, bindings) {
-  const el = e.currentTarget,
+function doMagic(e, bindings, testEl) {
```
Comment (davidandrewfrey) [inline behaviors/magic-button.js]: The name is 🌟 🌟 🌟 🌟 🌟

File: behaviors/magic-button.js (hunk 1 above)
Comment (davidandrewfrey) [inline behaviors/magic-button.js]: Sounds good.

File: behaviors/magic-button.js (hunk 3 above)
Comment (davidandrewfrey) [inline behaviors/magic-button.js]: Cool. I'll keep it in mind for consistency's sake. Feel like ternarys are fine as long as their either semi-complex AND only one level deep (like the second half would be below.) Or 2 levels deep if all moving parts are simple.

**Comment by davidandrewfrey** [comment]
👍

**Pull request merged**